### PR TITLE
Refactor rendering process

### PR DESCRIPTION
**Pull request by plopesc**
Refactor rendering process in 2 ways:

- Moving code in views_geojson file to the GeoJson class to improve encapsulation and have all the code together. Functions there are called only from that class.
- Refactoring row generation, removing foreach loops done in every row, doing a previous preprocess of the data and reusing results of that preprocess. I tested with a feed with 500 results and its faster.

Thanks

### CODE REVIEW

File: src/Plugin/views/style/GeoJson.php [hunk 1]
```diff
@@ -303,28 +305,27 @@ public function render() {
...
-      if ($feature = _views_geojson_render_fields($this->view, $row, $i)) {
+      if ($feature = $this->renderFields($row, $excluded_fields)) {
         $features['features'][] = $feature;
       }
     }
     unset($this->view->row_index);
 
     // Render the collection to JSON.
-    $json = \Drupal\Component\Serialization\Json::encode($features);
+    $json = Json::encode($features);
```
Comment (timstallmann) [inline src/Plugin/views/style/GeoJson.php]: @plopesc Drupal's `Json::encode()` is just a thin wrapper around `json_encode`, which we could just call directly with the `PRETTY_PRINT` flag, which has the nice advantage of making the `niceJson` method unnecessary. Thoughts?

Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann sounds good to me! but I haven't tested that, if works, go ahead, please :+1:

File: src/Plugin/views/style/GeoJson.php [hunk 2]
```diff
@@ -338,4 +339,246 @@ public function getFormats() {
...
+   *
+   * @param \Drupal\views\ResultRow $row
+   *   Row object
+   * @param array $excluded fields
+   *   array containing the list of fields to exclude from the rendering.
+   *
+   * @return array
+   *   Array containing all the raw and rendered fields
+   */
+  protected function renderFields(ResultRow $row, array $excluded_fields) {
```
Comment (timstallmann) [inline src/Plugin/views/style/GeoJson.php]: @plopesc in testing I'm getting an error that this overrides the default signature for `Drupal\views\Plugin\views\style\StylePluginBase::renderFields(array $result)` -- did you see that error locally?

Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann what PHP version are you using? I've tested with PHP7.

I'm not getting that error. But you can easily bypass it renaming the method to renderRow(), for instance :)

**Comment by timstallmann** [comment]
@plopesc opened up a new PR incorporating your changes plus a few other fixes -- would you mind taking a look? https://github.com/savaslabs/views_geojson/pull/49

**Pull request merged**